src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -325,6 +325,28 @@ internal virtual NamedTypeSymbol LookupMetadataType(ref MetadataTypeName emitted
                 }
             }
 
+            if (isTopLevel && MetadataHelpers.DecodeFileType(emittedTypeName.UnmangledTypeName) is (not -1 and var ordinal, var typeName))
```

Comment:
> if (isTopLevel && MetadataHelpers.DecodeFileType(emittedTypeName.UnmangledTypeName) is (not -1 and var ordinal, var typeName)) [](http://example.com/codeflow?start=11&length=127)

Is there a need to enter this ```if``` for ```PENamespaceSymbol```?

Reply:
I think we can add a check here to prevent having to do the additional lookup for `PENamespaceSymbol`.